includes/class-image-optimisation.php

```
@@ -251,6 +298,75 @@ private function is_valid_url( $url ) {
+			if ( empty( $url ) || strpos( $url, 'data:' ) === 0 ) {
+				return $url;
+			}
+
+			// Protocol-relative URLs (e.g., //example.com/image.jpg).
+			if ( strpos( $url, '//' ) === 0 ) {
+				$scheme = wp_parse_url( home_url(), PHP_URL_SCHEME ) ?: 'http';
+				return $scheme . ':' . $url;
+			}
+
+			// Root-relative paths (e.g., /wp-content/uploads/image.jpg).
+			if ( strpos( $url, '/' ) === 0 && strpos( $url, '//' ) !== 0 ) {
```

PHPStan (2.1.46)

[warning] Strict comparison using !== between int<1, max>|false and 0 will always evaluate to true.


(notIdentical.alwaysTrue)